lms/djangoapps/instructor_task/tasks_helper/misc.py

```diff
@@ -285,3 +288,56 @@ def upload_ora2_data(
     TASK_LOG.info(u'%s, Task type: %s, Upload complete.', task_info_string, action_name)
 
     return UPDATE_STATUS_SUCCEEDED
+
+
+def reset_progress_student(_xmodule_instance_args, _entry_id, course_id, _task_input, action_name):
+    start_time = time()
+
+    num_attempted = 1
+    num_total = 1
+
+    fmt = u'Task: {task_id}, InstructorTask ID: {entry_id}, Course: {course_id}, Input: {task_input}'
+    task_info_string = fmt.format(
+        task_id=_xmodule_instance_args.get('task_id') if _xmodule_instance_args is not None else None,
+        entry_id=_entry_id,
+        course_id=course_id,
+        task_input=_task_input
+    )
+    TASK_LOG.info(u'%s, Task type: %s, Starting task execution', task_info_string, action_name)
+
+    task_progress = TaskProgress(action_name, num_total, start_time)
+    task_progress.attempted = num_attempted
+
+    curr_step = {'step': "Creating user"}
+
+    TASK_LOG.info(
+        u'%s, Task type: %s, Current step: %s for all submissions',
+        task_info_string,
+        action_name,
+        curr_step,
+    )
+
+    user = User.objects.get(email=_task_input['student'])
+
+    new_user = User.objects.create(email="{}.{}".format(user.email, start_time),
+                                   username="{}.{}".format(user.username, start_time))
+    new_profile = user.profile
+    new_profile.pk, new_profile.user = None, new_user
+    new_profile.save()
+
+    task_progress.succeeded = 1
+
+    curr_step = {'step': "Reseting progress"}
+    TASK_LOG.info(
+        u'%s, Task type: %s, Current step: %s',
+        task_info_string,
+        action_name,
+        curr_step,
+    )
+    task_progress.update_task_state(extra_meta=curr_step)
+
```

Comment:
i'm ok with this.
@dpowell could you please check logic here also (i mean `reset_progress_student` function) ? to avoid situation if we forgot something .